### DISCUSSION TITLE
Executor timeout value

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1945] [Q&A] [answered]
Hey team,

I have an executor within my wave app to run a task in the background. Is there any timeout value for this executor by default or can we define it?

    with concurrent.futures.ThreadPoolExecutor() as executor:
        await q.exec(
            executor,
            run_aiem_query,
            *(
                q,
            ),
        )

cc: @mturoci

**Comment by mturoci** [comment, marked as answer]
Hi, there is no timeout imposed by Wave itself. See the [implementation](https://github.com/h2oai/wave/blob/48b5dc5495437148f5490ae1d6a46d67ebd7b3b9/py/h2o_wave/h2o_wave/server.py#L175). You can specify your own.

**Reply by shamilh2o** [reply]
Thanks @mturoci
